plugin/src/main/java/org/wildfly/swarm/plugin/gradle/GradleArtifactResolvingHelper.java

```diff
@@ -87,27 +77,25 @@ public ArtifactSpec resolve(final ArtifactSpec spec) {
             return specs;
         }
 
-        final Set<ArtifactSpec> resolvedSpecs = new HashSet<>();
-
-        doResolve(specs, transitive).forEach(dep -> dep.getModuleArtifacts()
-                .forEach(artifact -> resolvedSpecs
-                        .add(new ArtifactSpec(dep.getConfiguration(),
-                                dep.getModuleGroup(),
-                                artifact.getName(),
-                                dep.getModuleVersion(),
+        return doResolve(specs, transitive)
+            .stream()
+            .map(artifact -> new ArtifactSpec("default",
+                                artifact.getModuleVersion().getId().getGroup(),
+                                artifact.getModuleVersion().getId().getName(),
+                                artifact.getModuleVersion().getId().getVersion(),
                                 artifact.getExtension(),
                                 artifact.getClassifier(),
-                                artifact.getFile()))));
-
-        return resolvedSpecs.stream()
-                .filter(a -> !"system".equals(a.scope))
```

Comment:
Filtering out all system-scoped dependencies was unnecessary. All artifacts will have scope "default", the default-name of all detached configurations which the GradleArtifactResolvingHelper is using to resolve artifacts for the dependencies.

I think this is a leftover from the maven-implementation.